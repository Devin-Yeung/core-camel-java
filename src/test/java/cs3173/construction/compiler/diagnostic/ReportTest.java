package cs3173.construction.compiler.diagnostic;


import com.opencastsoftware.yvette.Position;
import org.approvaltests.Approvals;
import org.junit.Test;

import java.io.IOException;

public class ReportTest {


    @Test
    public void helloWorld() {
        Error err = Error.builder()
                .message("This is an Error Message")
                .start(new Position(0, 0))
                .end(new Position(0, 5))
                .hint("This is a Hint")
                .build();

        Reporter reporter = new Reporter("Hello World", false);
        reporter.report(err);
        try {
            Approvals.verify(reporter.display());
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}
