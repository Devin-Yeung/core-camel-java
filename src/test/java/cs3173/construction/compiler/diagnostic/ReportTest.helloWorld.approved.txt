
  × This is an Error Message
   ╭─[example.ml:1:1]
 1 │ Hello World
   · ──┬──
   ·   ╰── This is a Hint
   ╰────
